appcues/src/main/java/com/appcues/ui/AppcuesOverlayViewManager.kt

```diff
@@ -123,15 +132,59 @@ internal class AppcuesOverlayViewManager(private val scope: Scope) : DefaultLife
-    }
-
     private fun ViewGroup.setAccessibility(accessibilityFlag: Int) {
         children.forEach { child ->
             if (child.id != R.id.appcues_overlay_layout && child.id != R.id.appcues_debugger_view)
                 child.importantForAccessibility = accessibilityFlag
         }
     }
 }
+
+internal fun Activity.getParentView(): ViewGroup {
+
+    // try to find the most applicable decorView to inject Appcues content into. Typically there is just a single
+    // decorView on the Activity window. However, if something like a dialog modal has been shown, this can add another
+    // window with another decorView on top of the Activity. If we want to support showing content above that layer, we need
+    // to find the top most decorView like below.
+    
+    val decorView = if (VERSION.SDK_INT >= VERSION_CODES.Q) {
+        // this is the preferred method on API 29+ with the new WindowInspector function
+        WindowInspector.getGlobalWindowViews().last()
+    } else {
+        @Suppress("SwallowedException", "TooGenericExceptionCaught")
+        try {
+            // this is the less desirable method for API 21-28, using reflection to try to get the root views
+            val windowManagerClass = Class.forName("android.view.WindowManagerGlobal")
+            val windowManager = windowManagerClass.getMethod("getInstance").invoke(null)
+            val getViewRootNames: Method = windowManagerClass.getMethod("getViewRootNames")
+            val getRootView: Method = windowManagerClass.getMethod("getRootView", String::class.java)
+            val rootViewNames = getViewRootNames.invoke(windowManager) as Array<Any?>
+            val rootViews = rootViewNames.map { getRootView(windowManager, it) as View }
+            rootViews.last()
+        } catch (e: Exception) {
+            // if all else fails, use the decorView on the window, which is typically the only one
+            window.decorView
+        }
+    }
+
+    // This is the case of some other decorView showing on top - like a modal
+    // dialog. In this case, we need to apply the fix-ups below to ensure that our
+    // content can render correctly inside of this other view. In each case, we use
+    // the applicable value from the Activity default window.
+    if (ViewTreeLifecycleOwner.get(decorView) == null) {
+        val lifecycleOwner = window.decorView.findViewTreeLifecycleOwner()
+        ViewTreeLifecycleOwner.set(decorView, lifecycleOwner)
+    }
+
+    if (ViewTreeViewModelStoreOwner.get(decorView) == null) {
+        val viewModelStoreOwner = window.decorView.findViewTreeViewModelStoreOwner()
+        ViewTreeViewModelStoreOwner.set(decorView, viewModelStoreOwner)
+    }
+
+    if (decorView.findViewTreeSavedStateRegistryOwner() == null) {
+        val savedStateRegistryOwner = window.decorView.findViewTreeSavedStateRegistryOwner()
+        decorView.setViewTreeSavedStateRegistryOwner(savedStateRegistryOwner)
+    }
```

Comment:
workarounds for issues like those noted in https://issuetracker.google.com/issues/180691023 - we can't rely on the decorView to be well formed in some cases other than the normal main window decorView.

Reply:
good catch